services/vm-watcher/vm_watcher_service.go

```diff
@@ -144,7 +155,10 @@ func (s Service) vmFolderMoRefWithIDs(
 			}
 			moids[v] = append(moids[v], fmt.Sprintf("%s/%s", z.Namespace, z.Name))
 
-			if !controllerutil.ContainsFinalizer(&z, zonectrl.Finalizer) {
+			// API server prevents adding finalizers to resources that
+			// are in Deleting state, so skip such zones.
+			if z.DeletionTimestamp.IsZero() &&
```

Comment:
This second `IsZero()` check shouldn't be needed, right? With out it makes it clearer that for us to watch something - that is, its in the `moids` maps - the finalizer must be present.

Reply:
For posterity: without this, we are saying that regardless of whether the zone is in deleting or not, we will add the finalizer if it is not present.
If the zone is deleting, and it doesn't have the finalizer, we would have returned from L143.  And if it is not deleting, we want to add the finalizer.

So yeah, this seems redundant.  Removing.